csfieldguide/chapters/management/commands/loadchapters.py

```diff
@@ -49,10 +48,25 @@ def handle(self, *args, **options):
                 ["chapters"],
                 "Application Structure"
             )
-        for (chapter_slug, chapter_structure) in chapters.items():
-            factory.create_chapter_loader(
-                structure_file_path,
-                chapter_slug,
-                chapter_structure,
-                BASE_PATH
-            ).load()
+        else:
+            for chapter_slug in chapters:
+                chapter_structure_file_path = os.path.join(
+                    BASE_PATH,
+                    "{}/{}.yaml".format(chapter_slug, chapter_slug)
+                )
+                chapter_number = chapters[chapter_slug]["chapter-number"]
```

Comment:
Wouldn't this logic be better in the chapter loader?

Reply:
The chapter number is specified in the `structure.yaml` file (along with the chapter slug). Therefore it makes sense to have this code in this file because this is where `structure.yaml` is parsed.
I could move this to the chapter loader, but the number would still have to be extracted in loadchapters anyway, and additionally, the `structure.yaml` file path would have to be given as an additional parameter in order to be used in the error handling code for checking the chapter number is valid.
Overall, it seemed much cleaner to have this code in the loadchapters file.
Thoughts?

Reply:
My thoughts were thinking of where it would be loaded irrelevant to the source files. I thought of the logic in three parts:

1) Module for handling Django command call.
2) Module for loading chapters.
3) Module for loading chapter sections.

If I were to guess where the logic for the chapter number is located without looking at files, it would be in number 2.

Would it be better to move a topic's data to its own configuration file (like CS Unplugged's topics) so extra values like authors can be easily added later? Or would you rather have it as one longer file?

Reply:
@hayleyavw Pointed out in offline discussion that listing all chapter numbers together makes it easy for authors to see/modify order of all chapters. So happy to leave this as is.